src/mlpack/tests/main_tests/hoeffding_tree_test.cpp

```diff
@@ -0,0 +1,356 @@
+    BOOST_FAIL("Cannot load train dataset vc2.csv!");
+
+  arma::Row<size_t> labels;
+  if (!data::Load("vc2_labels.txt", labels))
+    BOOST_FAIL("Cannot load labels for vc2_labels.txt");
+
+  arma::mat testData;
+  if (!data::Load("vc2_test.csv", testData, info))
+    BOOST_FAIL("Cannot load test dataset vc2.csv!");
+
+  size_t testSize = testData.n_cols;
+
+  // Input training data.
+  SetInputParam("training", std::move(std::make_tuple(info, inputData)));
+  SetInputParam("labels", std::move(labels));
+
+  // Input test data.
+  SetInputParam("test", std::move(std::make_tuple(info, testData)));
+
+  mlpackMain();
+
+  // Check that number of output points are equal to number of input points.
+  BOOST_REQUIRE_EQUAL(
+      CLI::GetParam<arma::Row<size_t>>("predictions").n_cols, testSize);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("probabilities").n_cols,
+      testSize);
+
+  // Check number of output rows equals number of classes in case of
+  // probabilities and 1 for predictions.
+  BOOST_REQUIRE_EQUAL(
+      CLI::GetParam<arma::Row<size_t>>("predictions").n_rows, 1);
+  BOOST_REQUIRE_EQUAL(
+      CLI::GetParam<arma::mat>("probabilities").n_rows, 3);
+}
+
+/**
+ * Check that number of output points and number
+ * of input points are equal for categorical dataset.
+ */
+BOOST_AUTO_TEST_CASE(HoeffdingTreeCategoricalOutputDimensionTest)
+{
+  arma::mat inputData;
+  DatasetInfo info;
+  if (!data::Load("braziltourism.arff", inputData, info))
+    BOOST_FAIL("Cannot load train dataset braziltourism.arff!");
+
+  arma::Row<size_t> labels;
+  if (!data::Load("braziltourism_labels.txt", labels))
+    BOOST_FAIL("Cannot load labels for braziltourism_labels.txt");
+
+  arma::mat testData;
+  if (!data::Load("braziltourism_test.arff", testData, info))
+    BOOST_FAIL("Cannot load test dataset braziltourism_test.arff!");
+
+  size_t testSize = testData.n_cols;
+
+  // Input training data.
+  SetInputParam("training", std::move(std::make_tuple(info, inputData)));
+  SetInputParam("labels", std::move(labels));
+
+  // Input test data.
+  SetInputParam("test", std::move(std::make_tuple(info, testData)));
+
+  mlpackMain();
+
+  // Check that number of output points are equal to number of input points.
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Row<size_t>>
+      ("predictions").n_cols, testSize);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("probabilities").n_cols,
+      testSize);
+
+  // Check number of output rows equals number of classes in case of
+  // probabilities and 1 for predictions.
+  BOOST_REQUIRE_EQUAL(
+      CLI::GetParam<arma::Row<size_t>>("predictions").n_rows, 1);
+  BOOST_REQUIRE_EQUAL(
+      CLI::GetParam<arma::mat>("probabilities").n_rows, 6);
+}
+
+/**
+ * Check whether providing labels explicitly and extracting from last
+ * dimension give the same output.
+ */
+BOOST_AUTO_TEST_CASE(HoeffdingTreeLabelLessTest)
+{
+  arma::mat inputData;
+  DatasetInfo info;
+  if (!data::Load("tae.csv", inputData, info))
```

Comment:
Do you think we could reuse an existing dataset here e.g. thyroid or reconstruct the iris dataset, the repo size is already quite high, so I think whenever we can we should avoid adding datasets. Let me know what you think.

Reply:
Of course, a dataset with last column as labels was needed, but modifying iris would do the trick too. :+1: